au.org.access-nri/model/spack/environment/deployment/1-0-0.json

```diff
@@ -0,0 +1,307 @@
+                          ]
+                        }
+                      },
+                      {
+                        "type": "string"
+                      }
+                    ]
+                  },
+                  "version": {},
+                  "target": {
+                    "type": "array",
+                    "default": [],
+                    "items": {
+                      "type": "string"
+                    }
+                  },
+                  "compiler": {
+                    "type": "array",
+                    "default": [],
+                    "items": {
+                      "type": "string"
+                    }
+                  },
+                  "buildable": {
+                    "type": "boolean",
+                    "default": true
+                  },
+                  "permissions": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                      "read": {
+                        "type": "string",
+                        "enum": [
+                          "user",
+                          "group",
+                          "world"
+                        ]
+                      },
+                      "write": {
+                        "type": "string",
+                        "enum": [
+                          "user",
+                          "group",
+                          "world"
+                        ]
+                      },
+                      "group": {
+                        "type": "string"
+                      }
+                    }
+                  },
+                  "package_attributes": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "patternProperties": {
+                      "\\w+": {}
+                    }
+                  },
+                  "providers": {
+                    "type": "object",
+                    "default": {},
+                    "additionalProperties": false,
+                    "patternProperties": {
+                      "\\w[\\w-]*": {
+                        "type": "array",
+                        "default": [],
+                        "items": {
+                          "type": "string"
+                        }
+                      }
+                    }
+                  },
+                  "variants": {
+                    "oneOf": [
+                      {
+                        "type": "string"
+                      },
+                      {
+                        "type": "array",
+                        "items": {
+                          "type": "string"
+                        }
+                      }
+                    ]
+                  }
+                }
+              }
+            },
+            "patternProperties": {
+              "(?!^all$)(^\\w[\\w-]*)": {
```

Comment:
This matches any property that isn't `all`?

Reply:
Correct